docs/contributing.rst

```diff
@@ -0,0 +1,261 @@
+
+
+Type Hints
+~~~~~~~~~~
+
+This code base makes use of `type hints`_. Type hints make it easy to prevent
+certain types of bugs, enable richer tooling, and enhance the documentation,
+making the code easier to follow.
+
+All new code is required to include type hints, with the exception of tests.
+
+All parameters, as well as the return type of defs, are expected to be typed,
+with the exception of ``self`` and ``cls`` as seen in the following example.
+
+.. code:: python
+
+    def __init__(self, wrapped_db: DatabaseAPI) -> None:
+        self.wrapped_db = wrapped_db
+        self.reset()
+
+
+Documentation
+~~~~~~~~~~~~~
+
+Good documentation will lead to quicker adoption and happier users. Please
+check out our guide on `how to create documentation`_ for the Python Ethereum
+ecosystem.
+
+
+Pull Requests
+~~~~~~~~~~~~~
+
+It's a good idea to make pull requests early on. A pull request represents the
+start of a discussion, and doesn't necessarily need to be the final, finished
+submission.
+
+See GitHub's documentation for `working on pull requests`_.
+
+Once you've made a pull request take a look at the Circle CI build status in
+the GitHub interface and make sure all tests are passing. In general, pull
+requests that do not pass the CI build yet won't get reviewed unless explicitly
+requested.
+
+If the pull request introduces changes that should be reflected in the release
+notes, please add a `newsfragment` file as explained
+`here <https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md>`_
+
+If possible, the change to the release notes file should be included in the
+commit that introduces the feature or bugfix.
+
+
+Releasing
+~~~~~~~~~
+
+Final test before each release
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Before releasing a new version, build and test the package that will be released.
+There's a script to build and install the wheel locally, then generate a temporary
+virtualenv for smoke testing:
+
+.. code:: sh
+
+    $ git checkout master && git pull
+
+    $ make package
+
+    # in another shell, navigate to the virtualenv mentioned in output of ^
+
+    # load the virtualenv with the packaged web3.py release
+    $ source package-smoke-test/bin/activate
+
+    # smoke test the release
+    $ pip install ipython
+    $ ipython
+    >>> from web3.auto import w3
+    >>> w3.isConnected()
+    >>> ...
+
+
+Verify the latest documentation 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To preview the documentation that will get published:
+
+.. code:: sh
+
+    $ make docs
+
+
+Preview the release notes
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code:: sh
+
+   $ towncrier --draft
+
+
+Compile the release notes
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After confirming that the release package looks okay, compile the release notes:
+
+.. code:: sh
+
+    $ make notes bump=$$VERSION_PART_TO_BUMP$$
+
+
+You may need to fix up any broken release note fragments before committing. Keep
+running make build-docs until it passes, then commit and carry on.
```

Comment:
it might be helpful to wrap "make build-docs" in something to make the command obvious (aka `make build-docs`)